src/batch/src/task/task_manager.rs

```diff
@@ -100,13 +101,30 @@ impl BatchManager {
         }
     }
 
-    pub fn check_if_task_aborted(&self, task_id: &TaskId) -> Result<()> {
+    pub fn check_if_task_aborted(&self, task_id: &TaskId) -> Result<bool> {
         match self.tasks.lock().get(task_id) {
             Some(task) => task.check_if_aborted(),
             None => Err(TaskNotFound.into()),
         }
     }
 
+    pub async fn wait_until_task_aborted(&self, task_id: &TaskId) -> Result<()> {
```

Comment:
1. Add `#[test]` config to this method since it should only used for test.
2. I suggest to make it mod private rather pub method.